### PR TITLE
Dylan

### PR DESCRIPTION
**Pull request by PromAtlas**
Worked on the search, spell checker, and BingHelper classes/functions.

### CODE REVIEW

File: backend/src/BingHelper.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
 from azure.cognitiveservices.search.newssearch import NewsSearchClient
 from msrest.authentication import CognitiveServicesCredentials
 from etc.FileHelpers import *
-
+import requests
 class BingHelper:
```
Comment (WarmBuns) [inline backend/src/BingHelper.py]: nitpick: spacing

File: backend/src/SearchHelper.py [hunk 2]
```diff
@@ -1,41 +1,94 @@
...
+            self._filters[title] = BloomFilter(capacity=len(words)+self._buffer, error_rate=self._error_rate)
+            for word in words:
+                # If a word isn't in stopWords array, add it to the bloom filter.
+                if word in self._stop_words:
+                    pass
+                else:
+                    self._filters[title].add(word)
+                 
+    def spellCheck(self, search_string: str) -> []:
+
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py]: nitpick: spacing

File: backend/src/SearchHelper.py [hunk 3]
```diff
@@ -1,41 +1,94 @@
...
+    def spellCheck(self, search_string: str) -> []:
+        
+        response = (self.bingHelper.spellCheckerRequest(search_string)).json()
+        search_terms = re.split("\W+", search_string)
+        i = 0
+
+        if (len(response["flaggedTokens"]) == 0):
+            return search_terms
+        
+        #print(response)
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py]: delete

File: backend/src/SearchHelper.py [hunk 4]
```diff
@@ -1,41 +1,94 @@
...
+                else:
+                    self._filters[title].add(word)
+                 
+    def spellCheck(self, search_string: str) -> []:
+        
+        response = (self.bingHelper.spellCheckerRequest(search_string)).json()
+        search_terms = re.split("\W+", search_string)
+        i = 0
+
+        if (len(response["flaggedTokens"]) == 0):
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py:34]: comment on what this does

File: backend/src/SearchHelper.py [hunk 5]
```diff
@@ -1,41 +1,94 @@
...
-        #     except Exception:
-        #         print(f"Error finding synonyms on word: {word}")
-        
-        return synonyms
-
-    def _preprocess_search_query(self):
-        pass
\ No newline at end of file
+from pybloom import BloomFilter
+import pickle
```
Comment (Quikks1lver) [inline backend/src/SearchHelper.py:2]: more of a nitpick, but it's nice to have alphabetized imports

File: backend/src/SearchHelper.py [hunk 6]
```diff
@@ -1,41 +1,94 @@
...
+            return search_terms
+        
+        #print(response)
+        
+        for x in range(len(search_terms)):
+            if (search_terms[x] == response["flaggedTokens"][i]["token"]):
+                search_terms[x] = response["flaggedTokens"][i]["suggestions"][0]["suggestion"]
+                i += 1
+
+        #print(search_terms)
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py]: delete

File: backend/src/SearchHelper.py [hunk 7]
```diff
@@ -1,41 +1,94 @@
...
+        
+        return search_terms
+
+    # Assumes a dictionary of keyword-frequency dictionaries is passed in to work with
+    def search(self, search_string: str) -> str:
+
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py]: nitpick: spacing

File: backend/src/SearchHelper.py [hunk 8]
```diff
@@ -1,41 +1,94 @@
...
+                if (keyword in search_terms):
+                    for x in range(len(value)):
+                        result.append(value[x])
+
+        #print(result)
+        pickle_in.close()
+
+        return result
+
+
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py]: nitpick: too many spaces

File: backend/src/SearchHelper.py [hunk 9]
```diff
@@ -1,41 +1,94 @@
...
+                        result.append(value[x])
+
+        #print(result)
+        pickle_in.close()
+
+        return result
+
+    
+
+    def alt_search(self, search_string: str) -> List[str]:
```
Comment (WarmBuns) [inline backend/src/SearchHelper.py:64]: Is this the old search method?

File: frontend/package-lock.json [hunk 10]
```diff
@@ -3117,6 +3117,15 @@
       "resolved": "https://registry.npmjs.org/binary-extensions/-/binary-extensions-2.1.0.tgz",
       "integrity": "sha512-1Yj8h9Q+QDF5FzhMs/c9+6UntbD5MkRfRwac8DoEm9ZfUBZ7tZ55YcGVAzEe4bXsdQHEk+s9S5wsOKVdZrw0tQ=="
     },
```
Comment (WarmBuns) [inline frontend/package-lock.json]: This file shouldnt be comitted. Just delete it and push that and it will be removed from your PR

**Pull request merged**